kong/plugins/aws-lambda/iam-role-credentials.lua

```diff
@@ -0,0 +1,83 @@
+local http  = require "resty.http"
+local json  = require "cjson"
+local cache = require "kong.tools.database_cache"
+
+local CACHE_IAM_INSTANCE_CREDS_DURATION = 60 -- seconds to cache credentials from metadata service
+local IAM_CREDENTIALS_CACHE_KEY = "plugin.aws-lambda.iam_role_temp_creds"
+
+local function fetch_iam_credentials_from_metadata_service(metadata_service_host, metadata_service_port)
+    local client = http.new()
+    client:set_timeout(500)
```

Comment:
I get your concerns, I tried to omit any unnecessary config properties to keep the plugin configuration  relatively clean. Especially as the timeout would be another semi-optional config property (only applies if role based authentication is used). From my experience (using native AWS sdks on a variety of platforms) this is a property you would never need to touch. But maybe there are cases where you have too. 

I checked other "official" SDK's from Amazon. Most set a default timeout which cannot be changed. 
- [python cli](https://github.com/aws/aws-cli/blob/e2295b022db35eea9fec7e6c5540d06dbd6e588b/awscli/customizations/codedeploy/utils.py#L113) --> 1 second
- [python boto](https://github.com/boto/botocore/blob/cac78632cabddbc7b64f63d99d419fe16917e09b/botocore/session.py#L96) --> 1 second (can be configured)
- [java sdk](https://github.com/aws/aws-sdk-java/blob/7844c64cf248aed889811bf2e871ad6b276a89ca/aws-java-sdk-core/src/main/java/com/amazonaws/internal/ConnectionUtils.java#L41) --> 2 seconds

The timeouts from these SDK's are higher than the one set by me, but most of timeouts mentioned above apply for all services offered by the metadata service (and not just the credentials service). 

I'd suggest that I increase the timeout to 5 seconds by default and expose it as a parameter to `get_iam_credentials_from_instance_profile()`. - This way we can tweak it for the test and exposing it to the plugins config afterwards would be trivial if needed. 

If you want me to add it to the config I can do this of course too. Just let me know what you'd prefer.